From 1eab0b6a5a360910cb488fe9a964c058d3a0940a Mon Sep 17 00:00:00 2001
From: Chris Park <Chris.Park@amd.com>
Date: Thu, 26 Nov 2015 11:44:09 -0500
Subject: [PATCH 0525/1110] drm/amd/dal: Pass in Bios Parser instead of Adapter
 Service

In Stream Encoder to avoid calling the entire stack.

Signed-off-by: Harry Wentland <harry.wentland@amd.com>
Acked-by: Harry Wentland <harry.wentland@amd.com>
---
 drivers/gpu/drm/amd/dal/dc/dce110/dce110_resource.c       | 3 ++-
 drivers/gpu/drm/amd/dal/dc/dce110/dce110_stream_encoder.c | 5 ++---
 drivers/gpu/drm/amd/dal/dc/dce110/dce110_stream_encoder.h | 2 +-
 drivers/gpu/drm/amd/dal/include/stream_encoder_types.h    | 2 +-
 4 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_resource.c b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_resource.c
index 205cd22..ebae5e3 100644
--- a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_resource.c
+++ b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_resource.c
@@ -628,7 +628,8 @@ bool dce110_construct_resource_pool(
 		/* TODO: rework fragile code*/
 		enc_init_data.stream_engine_id = i;
 		enc_init_data.ctx = dc->ctx;
-		enc_init_data.adapter_service = adapter_serv;
+		enc_init_data.bp = dal_adapter_service_get_bios_parser(
+			adapter_serv);
 		if (pool->stream_engines.u_all & 1 << i) {
 			pool->stream_enc[i] = dce110_stream_encoder_create(
 					&enc_init_data);
diff --git a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_stream_encoder.c b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_stream_encoder.c
index 55be9f6..ac1a948 100644
--- a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_stream_encoder.c
+++ b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_stream_encoder.c
@@ -55,7 +55,7 @@ static void construct(
 {
 	enc->ctx = init->ctx;
 	enc->id = init->stream_engine_id;
-	enc->adapter_service = init->adapter_service;
+	enc->bp = init->bp;
 }
 
 static void update_avi_info_packet(
@@ -854,8 +854,7 @@ enum encoder_result dce110_stream_encoder_setup(
 	cntl.color_depth = crtc_timing->display_color_depth;
 
 	if (dal_bios_parser_encoder_control(
-			dal_adapter_service_get_bios_parser(
-			enc->adapter_service), &cntl) != BP_RESULT_OK)
+			enc->bp, &cntl) != BP_RESULT_OK)
 		return ENCODER_RESULT_ERROR;
 
 	return ENCODER_RESULT_OK;
diff --git a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_stream_encoder.h b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_stream_encoder.h
index 285d70e..d4477d1 100644
--- a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_stream_encoder.h
+++ b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_stream_encoder.h
@@ -28,7 +28,7 @@
 
 struct stream_enc_init_data {
 	enum engine_id stream_engine_id;
-	struct adapter_service *adapter_service;
+	struct bios_parser *bp;
 	struct dc_context *ctx;
 };
 
diff --git a/drivers/gpu/drm/amd/dal/include/stream_encoder_types.h b/drivers/gpu/drm/amd/dal/include/stream_encoder_types.h
index 0d3e67c..c0bc656 100644
--- a/drivers/gpu/drm/amd/dal/include/stream_encoder_types.h
+++ b/drivers/gpu/drm/amd/dal/include/stream_encoder_types.h
@@ -9,7 +9,7 @@
 
 struct stream_encoder {
 	enum engine_id id;
-	struct adapter_service *adapter_service;
+	struct bios_parser *bp;
 	struct dc_context *ctx;
 };
 
-- 
2.7.4

